R/bambu-extendAnnotations-utilityCombine.R

```diff
@@ -1,252 +1,219 @@
-                                   colDataCombined, rowData.unspliced, stranded)
-        se.combined <- SummarizedExperiment::rbind(se.spliced, se.unspliced)
-        rownames(se.combined) <- seq_len(nrow(se.combined))
-        return(se.combined)
-    }
+isore.combineTranscriptCandidates <- function(readClassList,
+    stranded, ## stranded used for unspliced reduce  
+    min.readCount = isoreParameters[["min.readCount"]], 
+    min.readFractionByGene = isoreParameters[["min.readFractionByGene"]],
+    bpParameters,
+    verbose){
+    combinedSplicedTranscripts <- 
+        combineSplicedTranscriptModels(readClassList, bpParameters, 
+        min.readCount, min.readFractionByGene, verbose)
+    combinedUnsplicedTranscripts <- 
+        combineUnsplicedTranscriptModels(readClassList,  bpParameters, 
+        stranded, min.readCount, min.readFractionByGene, verbose)
+    combinedTranscripts <- 
+        bind_rows(combinedSplicedTranscripts, combinedUnsplicedTranscripts)
+    return(combinedTranscripts)
 }
 
 
-#' create ref from a readClassSe object if readClassSeRef is not provided
-#' @importFrom dplyr %>% select
+#' combine spliced transcript models
+#' @noRd
+combineSplicedTranscriptModels <- function(readClassList, bpParameters, 
+    min.readCount = isoreParameters[["min.readCount"]], 
+    min.readFractionByGene = isoreParameters[["min.readFractionByGene"]],
+    min.geneFDR = isoreParameters[["min.geneFDR"]],
+    min.txFDR = isoreParameters[["min.txFDR"]],
+    verbose){
+    options(scipen = 999) #maintain numeric basepair locations not sci.notfi.
+    start.time <- proc.time()
+    featureTibbleList <- 
+        bplapply(seq_along(readClassList), function(sample_id){
+        extractFeaturesFromReadClassSE(readClassSe = readClassList[[sample_id]],
+        sample_id = sample_id)}, BPPARAM = bpParameters)
+    end.time <- proc.time()
+    if (verbose) message("creating spliced feature tibble objects for all 
+        samples in ", round((end.ptm - start.ptm)[3] / 60, 1)," mins.")
+    ## update combinedFeatureTibble by sample, as at each step 
+    ## start and end are updated iteratively as the fead count weighted average
+    start.time <- proc.time()
+    for ( s in seq_along(featureTibbleList)[-1]){
+        combinedFeatureTibble <- 
+        if_else(s==1, featureTibbleList[[s]],
+        bind_rows(list(combinedFeatureTibble,featureTibbleList[[s]]))) %>% 
+        group_by(intronStarts, intronEnds, chr, strand) %>%
+        mutate(start = median(rep(start, times = readCount)),
+            end = median(rep(end, times = readCount)), #weighted median is used 
+        NSampleReadCount = sum(readCount >= min.readCount), 
+        # number of samples passed read count criteria
+        NSampleReadProp = sum(geneReadProp >= min.readFractionByGene),
+        # number of samples passed gene read prop criteria (transcript usage)
+        NSampleGeneFDR = sum(geneFDR >= min.geneFDR),
+        NSampleTxFDR = sum(txFDR >= min.txFDR)) %>% 
```

Comment:
```suggestion
        NSampleGeneFDR = sum(geneFDR <= min.geneFDR),
        NSampleTxFDR = sum(txFDR <= min.txFDR)) %>% 
```
I put these in backwards, low FDR == better